hamilton/plugins/h_pydantic.py

```diff
@@ -0,0 +1,32 @@
+from typing import List
+
+from pydantic import BaseModel
+
+from hamilton import node
+from hamilton.data_quality import base as dq_base
+from hamilton.function_modifiers import InvalidDecoratorException
+from hamilton.function_modifiers import base as fm_base
+from hamilton.function_modifiers import check_output as base_check_output
+from hamilton.function_modifiers.validation import BaseDataValidationDecorator
+from hamilton.htypes import custom_subclass_check
+
+
+class check_output(BaseDataValidationDecorator):
+    def __init__(
+        self,
+        importance: str = dq_base.DataValidationLevel.WARN.value,
+        target: fm_base.TargetType = None,
+    ):
+        super(check_output, self).__init__(target)
+        self.importance = importance
+        self.target = target
+
+    def get_validators(self, node_to_validate: node.Node) -> List[dq_base.DataValidator]:
+        output_type = node_to_validate.type
```

Comment:
Oh, cool, really clever, I see what you did here. The reason this is it's own class is specifically so we can dynamically assign it based on the output type!

Was initially going to suggest you try to reuse `check_output_custom`, but this makes for a really nice UX. Mind adding some comments? It's really cool but a bit hidden as to why there's this class structure.

Musing -- I think the better way overall would be to expose the node to the subclass, but that's more work than it's worth now, and could have backwards compatibility. So yeah, some comments. 

And a docstring that we can put in docs!